src/internal/grpcutil/iter.go

```diff
@@ -0,0 +1,37 @@
+package grpcutil
+
+import (
+	"context"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/stream"
+	"google.golang.org/grpc"
+)
+
+type ClientStream[T any] interface {
+	Recv() (*T, error)
+	grpc.ClientStream
```

Comment:
I think this can be omitted.  If some other object also has Recv(*T, error), then let them be an iterator too :D

Reply:
We need Context(), and I don't know if I'm ready for this to be used for things that aren't gRPC, "some other object" should implement stream.Iterator.

Reply:
Fair.